src/dtls.c

```diff
@@ -1079,20 +1063,22 @@ static int DtlsCidGetSize(WOLFSSL* ssl, unsigned int* size, int rx)
     ConnectionID* id;
     CIDInfo* info;
 
-    if (ssl == NULL || size == NULL)
+    if (ssl == NULL)
         return BAD_FUNC_ARG;
 
     info = DtlsCidGetInfo(ssl);
     if (info == NULL)
         return WOLFSSL_FAILURE;
 
     id = rx ? info->rx : info->tx;
-    if (id == NULL) {
-        *size = 0;
-        return WOLFSSL_SUCCESS;
+    if (id == NULL || id->length == 0) {
+        if (size != NULL)
+            *size = 0;
+        return WOLFSSL_FAILURE;
```

Comment:
better to return SUCCESS in case of id->length == 0 as it's not an error case

Reply:
Restoring impl